frontend/class-frontend.php

```diff
@@ -1147,6 +1147,8 @@ public function metakeywords() {
 			$keywords = WPSEO_Meta::get_value( 'metakeywords' );
 			if ( $keywords === '' && ( is_object( $post ) && ( ( isset( $this->options[ 'metakey-' . $post->post_type ] ) && $this->options[ 'metakey-' . $post->post_type ] !== '' ) ) ) ) {
 				$keywords = wpseo_replace_vars( $this->options[ 'metakey-' . $post->post_type ], $post );
+			} else {
+				$keywords = wpseo_replace_vars( $keywords, $post );
 			}
```

Comment:
@andizer No worries, the logic behind it was difficult to explain, so can't have been easy to understand just by glancing at the code. Glad to hear it helped though.